lib/widgets/edit_state_marker.dart

```diff
@@ -25,11 +25,11 @@ class _SwipableMessageRowState extends State<SwipableMessageRow> {
   Widget build(BuildContext context) {
     final theme = Theme.of(context).extension<DesignVariables>()!;
 
-    // TODO(#157): fix how star marker aligns with message content
-    // Design from Figma at:
-    //   https://www.figma.com/file/1JTNtYo9memgW7vV6d0ygq/Zulip-Mobile?node-id=813%3A28817&mode=dev .
-    var star = Padding(padding: const EdgeInsets.only(top: 5.5),
-      child: Icon(ZulipIcons.star_filled, size: 16, color: theme.starColor));
+    var star = Padding(
+      padding: const EdgeInsets.only(top: 5.5),
+      child: Row(children: [
+          const Text("", style: TextStyle(fontSize: 15)),
+          Icon(ZulipIcons.star_filled, size: 16, color: theme.starColor)]));
```

Comment:
So normally, to align text-like elements by their baselines, we would use a `Row` with certain arguments passed:

```dart
Row(
  crossAxisAlignment: CrossAxisAlignment.baseline,
  textBaseline: localizedTextBaseline(context),
```

I don't know of a way to do that with a `Stack` instead of a `Row`, though. Do you think you could make it work—including with the animation—using a `Row`, where the first child is the edited/moved marker, the second is the message, and the third is the star? You may need to give the message area a fixed size, but I think that could be done with a `LayoutBuilder`, as in the diff below.

I haven't thoroughly tested this, and I don't know how it might interact with the animation requirement—but an advantage is that we'll keep the baseline alignment even as the message's first line varies in height. Screenshots below, showing a message starting with the line `# heading 1`, which makes a tall h1 element:

<details>
<summary>diff atop the PR branch</summary>

```diff
diff --git lib/widgets/edit_state_marker.dart lib/widgets/edit_state_marker.dart
index fdd40a43b..37fc93c7c 100644
--- lib/widgets/edit_state_marker.dart
+++ lib/widgets/edit_state_marker.dart
@@ -4,6 +4,7 @@ import 'package:flutter_gen/gen_l10n/zulip_localizations.dart';
 
 import '../api/model/model.dart';
 import 'icons.dart';
+import 'text.dart';
 import 'theme.dart';
 
 class SwipableMessageRow extends StatefulWidget {
@@ -25,26 +26,23 @@ class _SwipableMessageRowState extends State<SwipableMessageRow> {
   Widget build(BuildContext context) {
     final theme = Theme.of(context).extension<DesignVariables>()!;
 
-    var star = Padding(
-      padding: const EdgeInsets.only(top: 5.5),
-      child: Row(children: [
-          const Text("", style: TextStyle(fontSize: 15)),
-          Icon(ZulipIcons.star_filled, size: 16, color: theme.starColor)]));
     final hasMarker = widget.message.editState != MessageEditState.none;
 
-    return Stack(
-      children: [
-        if (hasMarker) Positioned(
 
-    var star = Padding(
-      padding: const EdgeInsets.only(top: 5.5),
-      child: Row(children: [
-          const Text("", style: TextStyle(fontSize: 15)),
-          Icon(ZulipIcons.star_filled, size: 16, color: theme.starColor)]));
     final hasMarker = widget.message.editState != MessageEditState.none;
 
-    return Stack(
-      children: [
-        if (hasMarker) Positioned(
-          left: 0,
-          child: EditStateMarker(editState: widget.message.editState)),
-        Padding(
-          padding: const EdgeInsets.only(left: 16, right: 16),
-          child: widget.child),
-        if (widget.message.flags.contains(MessageFlag.starred))
-          Positioned(
-            right: 0,
-            child: star),
-      ],
+    return LayoutBuilder(
+      builder: (context, constraints) => Row(
+        crossAxisAlignment: CrossAxisAlignment.baseline,
+        textBaseline: localizedTextBaseline(context),
+        children: [
+          hasMarker
+            ? EditStateMarker(editState: widget.message.editState)
+            : const SizedBox(width: 16),
+          SizedBox(width: constraints.maxWidth - 32, child: widget.child),
+          SizedBox(width: 16,
+            child: widget.message.flags.contains(MessageFlag.starred)
+              ? Icon(ZulipIcons.star_filled, size: 16, color: theme.starColor)
+              : null),
+        ],
+      ),
     );
   }
```
</details>

| This PR | with `Row` |
| --- | --- |
| ![E1B3BDED-0E58-425A-ABC3-7B646322F6BA](https://github.com/zulip/zulip-flutter/assets/22248748/0cb3a136-0425-48ad-9f42-a51b714f60d2) | ![CB5DDBAC-8715-44B2-92D0-1C26351110FE](https://github.com/zulip/zulip-flutter/assets/22248748/29460d13-6f7d-4677-ae7a-ec9934d15135) |

Reply:
https://github.com/zulip/zulip-flutter/assets/39874143/e5a74684-e444-4f80-bb70-6e48f44a3368

Wrapping the child with Transform.translate seems to work. We just need to deal with it overflowing the Row.